apps/spanishwotd/spanish_wotd.star

```diff
@@ -7,24 +7,48 @@ Author: logancornelius
 
 load("cache.star", "cache")
 load("encoding/json.star", "json")
-load("html.star", "html")
+load("re.star", "re")
 load("http.star", "http")
 load("render.star", "render")
 
-DEFAULT_WHO = "world"
-
-SPANISH_DICTIONARY_URL_1 = "https://www.dictionaryapi.com/api/v3/references/spanish/json/"
-SPANISH_DICTIONARY_URL_2 = "?key=d5ccb112-d698-45d6-ad07-c17a7b829b8d"
 CACHE_KEY = "wotd"
 CACHE_LLAVE = "pdd"
-CACHE_TTL = 10800  # 3 hours
-WOTD_CALENDAR_URL = "https://www.merriam-webster.com/word-of-the-day/calendar"  # Most succinct wotd definition
+CACHE_TTL = 3600  # 1 hour
+SPANISH_DICT_WOTD_URL = 'https://www.spanishdict.com/wordoftheday'
 
 def render_error():
     return render.Root(
         render.WrappedText("Something went wrong getting today's word!"),
     )
 
+def fetch_word_of_the_day():
+    wotd_resp = http.get(SPANISH_DICT_WOTD_URL)
+
+    if wotd_resp.status_code != 200:
+        return False
+
+    resp_body = wotd_resp.body()
+
+    pattern = r"window\.SD_COMPONENT_DATA\s*=(.*);"
+    matches = re.findall(pattern, resp_body)
+
+    if len(matches) == 0:
+        print("Failed to find word or definition from page")
+        return False
+
+    match = matches[0]
+
+    data = match.replace('window.SD_COMPONENT_DATA = ', '').replace(';', '')
+    parsed_data = json.decode(data)
+
+    wotd = parsed_data['wordOfTheDayData']
```

Comment:
the page also displays examples of using the word in a sentence, which could be fun to show as a marquee alongside the definition